src/app/(main)/connections/_components/modal.tsx

```diff
@@ -13,16 +11,14 @@ import {
 import { Input } from "@scylla-studio/components/ui/input";
 import { Modal } from "@scylla-studio/components/ui/modal";
 import { Plus } from "lucide-react";
-import { ReactNode, useState, useTransition } from "react";
+import React from "react";
+import { ReactNode, useEffect, useState, useTransition } from "react";
 import { FormProvider, useForm } from "react-hook-form";
 import { z } from "zod";
 
 const formSchema = z.object({
   name: z.string().min(1, { message: "Name is required." }),
-  host: z
-    .string()
-    .min(1, { message: "Host is required." })
-    .regex(/^[a-zA-Z0-9.-]+$/, "Invalid host format."),
+  host: z.string().ip(),
   username: z.string().min(1, { message: "Username is required." }),
```

Comment:
Since we're already going through this feature:

usernames usually follows the idea of `no spacing`, so this can be a rule to implement.